quesma/queryparser/pancake_pipelines.go

```diff
@@ -0,0 +1,178 @@
+// Copyright Quesma, licensed under the Elastic License 2.0.
+// SPDX-License-Identifier: Elastic-2.0
+package queryparser
+
+import (
+	"context"
+	"fmt"
+	"quesma/logger"
+	"quesma/model"
+	"quesma/model/bucket_aggregations"
+	"quesma/util"
+)
+
+type pancakePipelinesProcessor struct {
+	ctx context.Context
+}
+
+func (p pancakePipelinesProcessor) selectPipelineRows(pipeline model.PipelineQueryType, rows []model.QueryResultRow,
+	bucketAggregation *pancakeModelBucketAggregation) (
+	result []model.QueryResultRow) {
+
+	isCount := pipeline.IsCount()
+	for _, row := range rows {
+		newRow := model.QueryResultRow{Index: row.Index}
+		for _, col := range row.Cols {
+			if !isCount && bucketAggregation.isInternalNameCountColumn(col.ColName) {
+				continue
+			}
+			if !bucketAggregation.isInternalNameOrderByColumn(col.ColName) {
+				// we don't need order by (and actually it would break if we included them)
+				newRow.Cols = append(newRow.Cols, col)
+			}
+		}
+		fmt.Println()
+		result = append(result, newRow)
+	}
+	return
+}
+
+func (p pancakePipelinesProcessor) currentPipelineMetricAggregations(layer *pancakeModelLayer,
+	rows []model.QueryResultRow) (resultPerPipeline map[string]model.JsonMap) {
+
+	resultPerPipeline = make(map[string]model.JsonMap)
+
+	for _, pipeline := range layer.childrenPipelineAggregations {
+		typ := pipeline.queryType.PipelineAggregationType()
+		if typ != model.PipelineMetricsAggregation {
```

Comment:
Minor, but would inline `typ` here:
```suggestion
		if pipeline.queryType.PipelineAggregationType() != model.PipelineMetricsAggregation {
```